### PR TITLE
Aztec: Implements Calypso PreProcessor

### LINKED ISSUE #7446: Aztec: the post content should be pre-processed.
The post content should be pre-processed before using it with Aztec.

Our APIs don't provide standard-HTML, so Aztec can't handle it without some pre-processing to replace newlines with paragraphs and breaks.

The logic the preprocessor should implement is:

- Replace ranges delimited by `\n\n` with paragraphs.
- Replace single ocurrences of `\n` with `<br>` tags.
- Make sure nothing is replaced inside `<pre>` tags.

### PR DESCRIPTION
**Pull request by jleandroperez**
### Details:
In this PR we're implementing a Pre Processor that converts a Calypso-Generated Document into (hopefully) Valid HTML, which is (again, hopefully) Aztec Friendly.

Fixes #7446

### To test:
1. Make sure the `CalypsoProcessorTests` are green
2. Create a Post in Calypso Web, with multiple newlines + a pre element
3. Edit the Post created in (2) in Aztec
4. Make sure the world is a happy place again. With actual newlines and Paragraph Breaks.

Needs review: @diegoreymendez 
Thanks in advance Diego!

### CODE REVIEW

File: WordPress/WordPressTest/Aztec/CalypsoProcessorTests.swift [hunk 1]
```diff
@@ -0,0 +1,62 @@
+import XCTest
+@testable import WordPress
+
+
+// MARK: - CalypsoProcessorTests
+//
+class CalypsoProcessorTests: XCTestCase {
+
```
Comment (jleandroperez) [inline WordPress/WordPressTest/Aztec/CalypsoProcessorTests.swift]: Dear bot: I don't really like you

File: WordPress/Classes/ViewRelated/Aztec/Processors/CalypsoProcessor.swift [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    /// Returns the collection of Swift Ranges of all of the Pattern matches, in the specified String.
+    ///
+    private func rangesOfSubstrings(matching pattern: String, in text: String) -> [Range<String.Index>] {
+        guard let regex = try? NSRegularExpression(pattern: pattern, options: .caseInsensitive) else {
+            return []
+        }
+
+        let matches = regex.matches(in: text, options: [], range: text.foundationRangeOfEntireString)
+        let ranges = matches.flatMap { text.range(from: $0.range) }
+
```
Comment (jleandroperez) [inline WordPress/Classes/ViewRelated/Aztec/Processors/CalypsoProcessor.swift]: Same as above

File: WordPress/Classes/Extensions/String+Ranges.swift [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+            return nil
+        }
+
+        let utf16Start = UTF16Index(range.lowerBound)
+        let utf16End = UTF16Index(range.upperBound)
+
+        guard let start = Index(utf16Start, within: self), let end = Index(utf16End, within: self) else {
+            return nil
+        }
+
```
Comment (jleandroperez) [inline WordPress/Classes/Extensions/String+Ranges.swift]: And same as above, as well

**Comment by diegoreymendez** [comment]
Merging, since @jleandroperez is away right now.

**Pull request merged**